File: JAVA/1945. Sum of Digits of String After Convert.java
class Solution {
    public int getLucky(String s, int k) {
        StringBuffer str = new StringBuffer();
        for(int i=0;i<s.length();++i)
        {
            str.append(s.charAt(i) - 96);
        }
        int sum = 0;
        for(int i=0;i<str.length();++i)
        {
            sum += str.charAt(i) - '0';
        }
        while(k>1)
        {
            int temp = sum;
            sum = 0;
            while(temp>0)
            {
                sum += temp%10;
                temp /= 10;
            }
            k--;
        }
        return sum;
    }
}

File: C++/1945. Sum of Digits of String After Convert.cpp
class Solution {
public:
    int getLucky(string s, int k) {
        string converted = convert(s);
        int ans = 0;
        for(int i=0; i<k; i++){
            ans = transform(converted);
            converted = to_string(ans);
        }
        return ans;
    }
    string convert(string s){
        string temp = "";
        for(int i=0; i<s.size(); i++){
            int n = s[i]-'a'+1;
            temp += to_string(n);
        }
        return temp;
    }
    int transform(string s){
        int ans = 0;
        for(int i=0; i<s.size(); i++){
            int n = s[i]-'0';
            ans += n;
        }
        return ans;
    }
};
